api/v1alpha1/widget_dashboard_types.go

```diff
@@ -0,0 +1,60 @@
+package v1alpha1
+
+// A base template which users can "fork" and customize for their own dashboards.
+type BaseWidgetDashboardTemplate struct {
+	Name           string         `json:"name" yaml:"name"`                                   // The name of the dashboard template
+	DisplayName    string         `json:"displayName" yaml:"displayName"`                     // The display name of the dashboard template
+	TemplateConfig TemplateConfig `json:"templateConfig" yaml:"templateConfig"`               // The configuration of the dashboard template
+	FrontendRef    string         `json:"frontendRef,omitempty" yaml:"frontendRef,omitempty"` // The frontend reference for the dashboard template
+}
+
+// TemplateConfig defines the configuration for different screen sizes.
+type TemplateConfig struct {
+	Sm []WidgetTemplateConfigItem `json:"sm" yaml:"sm"` // Small screen configuration items
+	Md []WidgetTemplateConfigItem `json:"md" yaml:"md"` // Medium screen configuration items
+	Lg []WidgetTemplateConfigItem `json:"lg" yaml:"lg"` // Large screen configuration items
+	Xl []WidgetTemplateConfigItem `json:"xl" yaml:"xl"` // Extra large screen configuration items
+}
+
+// WidgetTemplateConfigItem represents a single widget's configuration within the grid.
+type WidgetTemplateConfigItem struct {
+	W    *int `json:"w" yaml:"w"`                           // The width of the widget in the grid
```

Comment:
Why have these fields as pointers?

Reply:
@justinorringer  good question. Because 0 is also a valid value. If these are not pointers and the value is 0, we will get a validation error because 0 is falsey. If we have a pointer, nil is falsey, but 0 is truthy.

But now looking at it, I see this is not a coordinate but a width, which would imply that 0 is BS. You don't want a widget with 0 width or height. Good catch! I will fix it.